### ISSUE TITLE
Steam controller no longer works after update 1481318740

### ISSUE THREAD

**Issue by antiBaconMachine** [ValveSoftware/steam-for-linux#4767]
#### Your system information

* Steam client version: 1481318740
* Distribution (e.g. Ubuntu): Ubuntu 16.10
* Opted into Steam client beta?: [No] 
* Have you checked for system updates?: [Yes]

Controller info:

Serial Number: FC541022AB
Dongle Serial Number: FD5380928B

#### Please describe your issue in as much detail as possible:

I've used a Steam Controller on 2 Ubuntu systems with no problems for a few months. Yesterday an update came through, now the controller does not function on either system, by which I mean it lights up but does not control anything on up to date Ubuntu clients. Windows on same hardware is still fine. I was able to get the following reproduction on the second machine which had not yet been updated:

#### Steps for reproducing this issue:

1. opened Steam 1476379980 in offline mode
2. Connected steam controller using wireless dongle
3. Verified working as expected
4. Restarted in online mode which immediately triggered update  to 1481318740
5. Controller now does nothing.

So tl;dr; seems like a regression somewhere between  1476379980 and  1481318740.

Things I have tried:

* Opting into beta (1481930308) - no change
* Operating the steam controller on windows and steam link - works as expected
* Selecting search for steam controller and holding down x and powering on - Light on controller pulses for 15 seconds or so then turns off
* Connecting via USB - limited functionality achieved, seems to function as a mouse but not actually work in any game

**Comment by kisak-valve** [comment]
Hello @antiBaconMachine, does the steam controller work with the updated udev rule at https://steamcommunity.com/app/353370/discussions/0/490123197956024380/?

You may need to reload your udev rules, then replug the wireless dongle or reboot after making that kind of change.

**Comment by antiBaconMachine** [comment]
Yes! That seems to do the trick.

Any idea why I never needed those rules before?

In any case thanks so much @kisak-valve

**Comment by williamblevins** [comment]
Issue also seen on Linux Debian under Steam 1482202190 and Beta 1482214037. Full controller functionality on Desktop and in Steam Big Picture. Mouse works via Steam Controller in game for about 5 seconds, then the only functioning button is the Steam button itself. Steam Controller functionality equivalent using USB dongle and direct connection.

Using udev rule:
```
# This rule is needed for basic functionality of the controller in Steam and keyboard/mouse emulation
SUBSYSTEM=="usb", ATTRS{idVendor}=="28de", MODE="0666"

# This rule is necessary for gamepad emulation
KERNEL=="uinput", MODE="0660", GROUP="<insert_mygroup_here>", OPTIONS+="static_node=uinput"

# Valve HID devices over USB hidraw
KERNEL=="hidraw*", ATTRS{idVendor}=="28de", MODE="0666"

# Valve HID devices over bluetooth hidraw
KERNEL=="hidraw*", KERNELS=="*28DE:*", MODE="0666"
```

Suggestions? Additional information needed?

**Comment by williamblevins** [comment]
Apparently, you need the parts marked as DualShock 4 also which is not obvious. Without those sections, Steam Controller will not work in game. Why are these necessary?

**Comment by williamblevins** [comment]
It seems to be working now though I am not sure why it wasn't working previously unless I have devices competing. After plugging in the controller I get 5 entries (0-4) with 0666 mask, but I also have 5-8 which are there with/without the controller plugged in.

**Comment by williamblevins** [comment]
5-8 are apparently mice and keyboard according to dmesg, so no mystery there. Again, not sure. Thanks.

**Comment by kisak-valve** [comment]
Closing as the original issue appears to have been resolved.
